x/nexus/keeper/general_message.go

```diff
@@ -10,13 +10,22 @@ import (
 	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
 )
 
-func getMessageKey(chain exported.ChainName, id string) key.Key {
+func getMessageKey(chain exported.ChainName, status exported.GeneralMessage_Status, id string) key.Key {
 	return generalMessagePrefix.
 		Append(key.From(chain)).
+		Append(key.From(status)).
 		Append(key.FromStr(id))
 }
 
-// SetNewMessage sets the given general message
+func (k Keeper) GetGeneralMessageID(ctx sdk.Context, sourceTxID string, sourceChain exported.ChainName) string {
+	counter := utils.NewCounter[uint](generalMessageNonceKey, k.getStore(ctx))
+	nonce := counter.Incr(ctx)
+
+	id := fmt.Sprintf("%s-%s-%x", sourceTxID, sourceChain, nonce)
```

Comment:
This is probably not necessary. Came about because of our discussion around the uniqueness of tx hashes across different evm chains, but the nonce ensures the ID is unique, even if hash is the same.

Reply:
removed source chain from the ID